loopdb/migrate.go

```diff
@@ -0,0 +1,404 @@
+package loopdb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	ErrLoopOutsNotEqual        = errors.New("loop outs not equal")
+	ErrLoopInsNotEqual         = errors.New("loop ins not equal")
+	ErrLiquidityParamsNotEqual = errors.New("liquidity params not equal")
+)
+
+// MigratorManager is a struct that handles migrating data from one SwapStore
+// to another.
+type MigratorManager struct {
+	fromStore SwapStore
+	toStore   SwapStore
+}
+
+// NewMigratorManager creates a new MigratorManager.
+func NewMigratorManager(fromStore SwapStore,
+	toStore SwapStore) *MigratorManager {
+
+	return &MigratorManager{
+		fromStore: fromStore,
+		toStore:   toStore,
+	}
+}
+
+// RunMigrations runs the migrations from the fromStore to the toStore.
+func (m *MigratorManager) RunMigrations(ctx context.Context) error {
+
+	log.Infof("Migrating loop outs...")
+
+	// Migrate loop outs.
+	err := m.migrateLoopOuts(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Checking loop outs...")
+
+	// Check that the loop outs are equal.
+	err = m.checkLoopOuts(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Migrating loop ins...")
+
+	// Migrate loop ins.
+	err = m.migrateLoopIns(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Checking loop ins...")
+
+	// Check that the loop ins are equal.
+	err = m.checkLoopIns(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Migrating liquidity parameters...")
+
+	// Migrate liquidity parameters.
+	err = m.migrateLiquidityParams(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Checking liquidity parameters...")
+
+	// Check that the liquidity parameters are equal.
+	err = m.checkLiquidityParams(ctx)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Migrations complete!")
+
+	return nil
+}
+
+func (m *MigratorManager) migrateLoopOuts(ctx context.Context) error {
+	// Fetch all loop outs from the fromStore.
+	loopOuts, err := m.fromStore.FetchLoopOutSwaps(ctx)
+	if err != nil {
+		return err
+	}
+
+	// For each loop out, create a new loop out in the toStore.
+	for _, loopOut := range loopOuts {
+		err := m.toStore.CreateLoopOut(ctx, loopOut.Hash, loopOut.Contract)
```

Comment:
In order to speed this up, we want to be able to batch all the insertion here. Otherwise, you're doing a single db transaction for each swap. Instead a series of swaps can be created, within a single DB transaction.